### DISCUSSION TITLE
Json keys fillable is ignoring

### DISCUSSION THREAD

**Discussion by projct1** [laravel/framework#53881] [Q&A] [answered]
Please tell me where I'm going wrong? I persistently created a issue in the laravel/framework repository, but they ignore me.

```php
class MyModel extends Model
{
    protected $fillable = [
        'name',
        'data->some_value'
    ];
}

$model = MyModel::create([
    'name' => 'Some name',
    'data' => ['some_value' => 1]
]);

//got `null` (`$fillable` ignored)
dd($model->data);
```

Its already is [documented](https://laravel.com/docs/11.x/eloquent#mass-assignment-json-columns), must works, but not working =\

**Comment by carlosmintfan** [comment]
This should be resolved as of https://github.com/laravel/framework/issues/53878#issuecomment-2541542727: "you must use the key exactly the same in the fill array with '->'..." so you must use `'data->some_value' => 1` instead of `'data' => ['some_value' => 1]`.

**Reply by projct1** [reply]
> This should be resolved as of [#53878 (comment)](https://github.com/laravel/framework/issues/53878#issuecomment-2541542727): "you must use the key exactly the same in the fill array with '->'..." so you must use `'data->some_value' => 1` instead of `'data' => ['some_value' => 1]`.

Thanks, but this is absurd :D

**Reply by ghost** [reply]
It is how they implemented it. Sadly the documentation is not so clear about it... Anyway I think that this has some limitations in the create regard...

**Reply by carlosmintfan** [reply]
@marius-mcp You could add a PR to the docs detailing how to use it or maybe I will do it.

**Reply by ghost** [reply]
@carlosmintfan  I don't have the project forked, so please feel free to do it.  I would be interested to see examples for both create and update because on create the 'col->key' seems to me that is not usable without 'col'.

**Reply by macropay-solutions** [reply, marked as answer]
@marius-mcp It could be if you put there all the allowed keys and you create with each of those allowed keys 
'col->key'  => 1,
'col->key2' => 2, 
instead of creating with a single 'col' => json or ['key' => 1, 'key2' =>2]
